.github/workflows/docker.yml

```diff
@@ -0,0 +1,118 @@
+name: Docker
+on: [workflow_dispatch, push]
```

Comment:
```suggestion
on: [workflow_dispatch, pull_request]
```
I think using `pull_request` here should trigger the action in the PR and we can try all of this in isolation from the `main` branch. I am not sure if having both means it is a job that needs manual trigger additionally to a PR, but that would be favourable.